core/src/main/java/com/google/zxing/common/CharacterSetECI.java

```diff
@@ -60,16 +61,23 @@ public enum CharacterSetECI {
   GB18030(29, "GB2312", "EUC_CN", "GBK"),
   EUC_KR(30, "EUC-KR");
 
+  // only character sets supported by the current JVM are registered here
+  // some JVMs, for example, do not support character sets ISO-8859-6/8/10/11/14
+  // restricted JVMs can be simulated with the "zxing.remove.eci.charsets" property
   private static final Map<Integer,CharacterSetECI> VALUE_TO_ECI = new HashMap<>();
   private static final Map<String,CharacterSetECI> NAME_TO_ECI = new HashMap<>();
   static {
+    String removeProp = System.getProperty("zxing.remove.eci.charsets", "");
```

Comment:
Instead of special-casing tests, I wonder if we should just test somehow for support of each encoding? It might be some overhead to check every single one when most are almost never used; maybe smarter to just catch and exception somewhere and abort. I'm a little hesitant to test differently from how the library would run.

Reply:
With the proposed change, `CharacterSetECI` is both checking for support of each encoding (via `Charset.isSupported()`), *and* also allowing an override via the new property. Is this what you meant? Or were you referring to possibly checking for support of each encoding in the unit tests and adjusting test assertions according to the runtime being used to build? This second option seemed a bit more brittle to me, as opposed to just having all builds on all JVMs simulate a more restricted JVM (via the new property).

Reply:
I could also remove the override property completely, and adjust `HighLevelEncodeTestCase.testECIs()` to assume that ECI 8 (ISO-8859-6) is supported on the JVM running the test -- this test was added after ECI 8 support was removed, so currently assumes the use of ECI 24 (Cp1256) in its place.

Reply:
OK. I would just skip the system property. If it can support whatever the JVM supports without failing otherwise, that's the goal and that's fine.

Reply:
OK, I think we're good to go. Please have a look at the latest diff and let me know if you have any other concerns. Thanks!